src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs

```diff
@@ -0,0 +1,113 @@
+        {
+            _buyerOrganisationRoleId = "RO98";
+            _odsApiBaseUrl = odsApiBaseUrl;
+        }
+
+        public async Task<OdsOrganisation> GetBuyerOrganisationByOdsCodeAsync(string odsCode)
+        {
+
+            var response = await _odsApiBaseUrl
+                .AppendPathSegment("organisations")
+                .AppendPathSegment(odsCode)
+                .AllowHttpStatus("3xx,4xx")
+                .GetJsonAsync<OdsResponse>();
+
+            var odsOrganisation = response.Organisation;
+
+            return odsOrganisation is null ? null : new OdsOrganisation
+            {
+                OrganisationName = odsOrganisation.Name,
+                OdsCode = odsCode,
+                PrimaryRoleId = _buyerOrganisationRoleId,
+                Address = OdsResponseAddressToAddress(odsOrganisation.GeoLoc.Location),
+                IsActive = IsActive(odsOrganisation),
+                IsBuyerOrganisation = IsBuyerOrganisation(odsOrganisation)
+            };
+        }
+
+        private static bool IsActive(OdsResponseOrganisation organisation)
+        {
+            return organisation.Status.Equals("Active", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private bool IsBuyerOrganisation(OdsResponseOrganisation organisation)
+        {
+            return organisation.Roles.Role.Any(role =>
+                 role.id.Equals(_buyerOrganisationRoleId, StringComparison.OrdinalIgnoreCase));
+        }
+
+        private static Address OdsResponseAddressToAddress(OdsResponseAddress odsAddress)
+        {
+            return new Address
+            {
+                Line1 = odsAddress.AddrLn1,
+                Line2 = odsAddress.AddrLn2,
+                Line3 = odsAddress.AddrLn3,
+                Line4 = odsAddress.AddrLn4,
+                Town = odsAddress.Town,
+                County = odsAddress.County,
+                Postcode = odsAddress.PostCode,
+                Country = odsAddress.Country,
+            };
+        }
+
+        private class OdsResponse
+        {
+            public OdsResponseOrganisation Organisation { get; set; }
+        }
+
+        private class OdsResponseOrganisation
+        {
+            public string Name { get; set; }
+
+            public string Status { get; set; }
+
+            public GeoLoc GeoLoc { get; set; }
+
+            public OdsResponseRoles Roles { get; set; }
+        }
+
+        private class GeoLoc
+        {
+            public OdsResponseAddress Location { get; set; }
+        }
+
+        private class OdsResponseAddress
+        {
+            public string AddrLn1 { get; set; }
+            public string AddrLn2 { get; set; }
+            public string AddrLn3 { get; set; }
+            public string AddrLn4 { get; set; }
+            public string Town { get; set; }
+            public string County { get; set; }
+            public string PostCode { get; set; }
+            public string Country { get; set; }
```

Comment:
Is it worth applying consistent formatting here? In your other POCOs you have white space between each property. I'll leave it up to you again.

Reply:
👍